server/subscribe.go

```diff
@@ -103,13 +103,12 @@ func (p *Plugin) getChannelsSubscribed(jwh *JiraWebhook) ([]string, error) {
 	if err != nil {
 		return nil, err
 	}
+	eventEnums := jwh.toEventEnums()
 
-	subIds := subs.Channel.IdByEvent[jwh.WebhookEvent]
+	subIds := subs.Channel.ById
```

Comment:
I unresolved the earlier conversation so you can quickly find it (also just a minor point: I'm not sure if self-resolving discussions is good PR-etiquette or not; I would guess it's not, since it hides what the reviewer asked about before they have a chance to see what the response was. See what I mean? I never self-resolve for that reason.  I had to expand each of the 'resolved' conversations to remember what we were talking about.)

So, if I understand correctly (and please correct me if I'm wrong), the problem is that some event types can belong to multiple groups. Event type `Issue Updated - Description` belongs to `Issue Updated - Description` and `Issue Updated` and `Issue Updated - All` -- something like that, right?

I guess I don't understand: why can't we have one map that maps every issue type (the actual issue type that we get from the webhook) to the channels that have subscribed to it. So, if a channel subscribes to `Issue Updated - All` then its id gets put into every `Issue Updated` and `Issue Updated - <field>` list. Then if it subscribes to `Issue Updated - Description` its id gets added (again) to `Issue Updated - Description`. (Just un-dupe when you get the channelId list.) Then if it unsubscribes to `Issue Updated - All` /one/ of its ids gets removed from all of the `Issue Updated` lists, leaving its id still in the `Issue Updated - Description`, which matches their subscription.

I'm probably missing something, because it seems doable to me. If I am, can you give the example?  Thanks!

(And great work!)

Reply:
>  I'm not sure if self-resolving discussions is good PR-etiquette or not

Thanks for pointing this out, you are absolutely right. Conversations should be left open until they are truly resolved.

> I guess I don't understand: why can't we have one map that maps every issue type (the actual issue type that we get from the webhook) to the channels that have subscribed to it. So, if a channel subscribes to Issue Updated - All then its id gets put into every Issue Updated and Issue Updated - <field> list. Then if it subscribes to Issue Updated - Description its id gets added (again) to Issue Updated - Description

I understand your point, and that method would certainly work. But I don't see the value in maintaining this map, as it will become cumbersome to keep up-to-date, and can result in bugs if it is not carefully kept up-to-date. We can run into issues as more fields are added, including user-defined custom fields.

What is the main benefit of using the map? Currently the `getChannelsSubscribed` function is the only place reading from the map, and it doesn't seem necessary for the purposes of this function. If the reason is for performance, I don't see much of a performance increase by initially filtering by event type. I'm not opposed to using the map, but I need to understand why this is the route we should take. Do you mind if we discuss this together as a team?

Reply:
The only concern I have is that we could be processing many many webhook events on a big installation (like, thousands/minute). I don't want to be creating a new list of what channels should be updated for every webhook. I'd rather do that work once in a while when a channel is subscribed/unsubscribed. That's pretty much the reasoning for me.

As for keeping the channel subscription up-to-date, is there any other place that the subscription would change, other than in the channel settings modal? Because that seems like a 1-1 mapping. On save: remove that channel's id from every list, then re-add them based on what the user just chose.

Reply:
But, come to think of it, that seems like a bit out of scope for this. Let's make that an improvement ticket.